### PR TITLE
Add decorated functions

### PR DESCRIPTION
**Pull request by d2lam**
Add decorated functions:
- `_decorateUrl`
- `_decorateAuthor`
- `_decorateCommit`

Also remove breaker mock and mock request directly instead

Relates to https://github.com/screwdriver-cd/screwdriver/issues/255

### CODE REVIEW

**Comment by d2lam** [comment]
Cool https://github.com/screwdriver-cd/screwdriver/issues/274 is resolved. This PR is good for review

File: test/index.test.js [hunk 1]
```diff
@@ -88,77 +66,61 @@ describe('index', () => {
...
             return scm.parseUrl({
                 checkoutUrl: 'https://batman@bitbucket.org/batman/test.git#mynewbranch',
                 token: 'myAccessToken'
             })
-            .then(() => assert.fail('Should not get here'))
-            .catch((error) => {
-                assert.calledWith(breakRunMock.runCommand, expectedOptions);
-                assert.deepEqual(error, err);
-            });
+                .then(() => assert.fail('Should not get here'))
```
Comment (FenrirUnbound) [inline test/index.test.js]: Note:

`.then()` takes 2 functions as arguments: the first one is invoked if the previous promise in the chain resolves and the second one is invoked if the previous promise in the chain rejects.  You could combine the `.catch()` statement into this `.then()` call so your test will error out with the correct message.

File: index.js [hunk 2]
```diff
@@ -131,6 +143,107 @@ class BitbucketScm extends Scm {
...
+                    url: body.links.html.href,
+                    name: body.display_name,
+                    username: body.username,
+                    avatar: body.links.avatar.href
+                };
+            });
+    }
+
+    /**
+   * Decorate a given SCM URI with additional data to better display
```
Comment (FenrirUnbound) [inline index.js]: Minor -- Spacing

File: test/index.test.js (hunk 1 above)
Comment (d2lam) [inline test/index.test.js]: Yeah just different styles. I did it that way before but Min thought it was hard to read so I switched to this.

**Pull request merged**